distributed/worker.py

```diff
@@ -2857,7 +2857,7 @@ def get_worker():
         return thread_state.execution_state["worker"]
     except AttributeError:
         try:
-            return first(Worker._instances)
+            return first(w for w in Worker._instances if w.status != "closed")
```

Comment:
Are there other statuses that should be filtered out? Or maybe we could use a whitelist of "good" statuses instead?

Reply:
For instance we could have:

```python
            return first(w for w in Worker._instances if w.status == "running")
```

or:

```python
            return first(w for w in Worker._instances if w.status not in ("closing", "closed"))
```

Reply:
Apparently the 4 possible statuses are `None`, running, closing and closed.

Reply:
Yes, using `"running"` should be OK.  The `"closing"` and `None` states are both very short-lived and so this is unlikely to be an issue for any particular case, but of course when you do this thousands of times it's nicer to be safe :)